telegram/contrib/botan.py

```diff
@@ -44,3 +46,23 @@ def track(self, message, event_name='event'):
         except URLError as error:
             self.logger.warn('Botan track error ' + str(error.reason))
             return False
+
+    def shorten(self, url, user_id):
+        if not (url.startswith('http://') or url.startswith('https://')):
+            url = 'http://' + url
+        try:
+            url2 = self.shorten_template.format(token=str(self.token),
+                                                user_ids=str(user_id),
+                                                url=str(url))
+            request = Request(url2,
+                              headers={'Content-Type': 'application/json'})
+            with urlopen(request) as response:
+                shortened_url = response.read()
+            return shortened_url.decode('utf-8')
+        except HTTPError as error:
+            self.logger.warn('Botan track error ' + str(error.code) + ':' + error.read().decode(
+                'utf-8'))
```

Comment:
please don't use string concat here, but rather the `logging` dict formatting:
`self.logger.warn('some text %s:%s', error.code, error.read().decode('utf-8')`